File: api-signature/csharp/src/SignatureException.cs
using System;
using tr.gov.tubitak.uekae.esya.api.common;

namespace tr.gov.tubitak.uekae.esya.api.signature
{
    /**
 * Base exception class for signature API related errors.
 * @author ayetgin
 */
    public class SignatureException : Exception
    {
        public SignatureException()
        {
        }

        public SignatureException(String aMessage)
            : base(aMessage)
        {
        }
        
        public SignatureException(Exception aCause)
            : base(null,aCause)
        {
        }

        public SignatureException(String aMessage, Exception aCause)
            : base(aMessage, aCause)
        {
        }
    }
}

File: api-signature/src/tr/gov/tubitak/uekae/esya/api/signature/SignatureException.java
package tr.gov.tubitak.uekae.esya.api.signature;

/**
 * Base exception class for signature API related errors.
 * @author ayetgin
 */
public class SignatureException extends Exception
{
    public SignatureException()
    {
    }

    public SignatureException(String aMessage)
    {
        super(aMessage);
    }

    public SignatureException(Throwable aCause)
    {
        super(aCause);
    }

    public SignatureException(String aMessage, Throwable cause)
    {
        super(aMessage, cause);
    }


}
